compiler/code-gen/declarations.cpp

```diff
@@ -1177,14 +1190,30 @@ void ClassMembersDefinition::compile_msgpack_deserialize(CodeGenerator &W, Class
   //}
   //
 
+  // See `compile_msgpack_serialize()` note for forward_list.
   std::vector<std::string> cases;
-  klass->members.for_each([&](ClassMemberInstanceField &field) {
+
+  cases.emplace_back("default: break;");
+
+  ClassPtr the_klass = klass;
+
+  // Put class' fields along with all parent's fields.
+  while (the_klass) {
+    std::vector<std::string> cases_inner;
+    the_klass->members.for_each([&](ClassMemberInstanceField &field) {
     if (field.serialization_tag != -1) {
-      cases.emplace_back(fmt_format("case {}: elem.convert(${}); break;", field.serialization_tag, field.var->name));
+      cases_inner.emplace_back(fmt_format("case {}: elem.convert(${}); break;", field.serialization_tag, field.var->name));
     }
-  });
+    });
 
-  cases.emplace_back("default: break;");
+    for (auto &i : cases_inner) {
+      cases.emplace_back(std::move(i));
+    }
+
+    the_klass = the_klass->parent_class;
+  }
+
+  std::reverse(cases.begin(), cases.end());
```

Comment:
Why do we need reverse?

Reply:
I've refactored this part slightly since your review.

The reverse is required since classes are traversed in reverse order (from children to parents), but we want fields to be dumped in natural order -- from parents to children.

That's why at least one reverse is required.